packages/gatsby/src/schema/schema.js

```diff
@@ -299,11 +299,15 @@ const mergeTypes = ({
   parentSpan,
 }) => {
   // Only allow user or plugin owning the type to extend already existing type.
+  // Themes are also allowed to modify plugin-owned types, but not types already
+  // modified by another plugin.
   const typeOwner = typeComposer.getExtension(`plugin`)
   if (
     !plugin ||
+    plugin.name === typeOwner ||
     plugin.name === `default-site-plugin` ||
-    plugin.name === typeOwner
+    (plugin.name.startsWith(`gatsby-theme-`) &&
+      !typeOwner.startsWith(`gatsby-theme-`))
```

Comment:
Just wonder that does the name of a theme have to starts with "gatsby-theme" ?  If so, I  think that's a little unreasonable. My theme plugin's is `antdsite` , not starts with "gatsby-theme". I think we should judge theme in another way.

Reply:
the `gatsby-theme-` naming is a requirement since #16620, so we can rely on that. it's not totally optimal since i don't think it is actually enforced anywhere, otoh the only thing you'll lose (currently) when naming your theme differently is the ability to modify types owned by plugins. i'm open to other proposals, please post them in #16529.

Reply:
Lol All right, since you haved discussed.

Reply:
sorry, looks like my assumption was wrong